### DISCUSSION TITLE
Bug in autoformat of markdown links.

### DISCUSSION THREAD

**Discussion by victor-chebotar** [GitbookIO/community#368] [3. Git Sync]
1. create external `.md` file
2. add markdown style link `[title](link-example.com?field_name=5)`
3. upload the file to the documentation
4. edit file using gitbook UI
5. commit diffs

actual
6. link is edited and looks like `[title](link-example.com?field\_name=5)` in source file

expected:
6. format normalization **doesn't escape the underscore** in markdown link syntax.

here is an example of the link that becomes edited:
[unix process model](https://adam.herokuapp.com/past/2011/5/9/applying_the_unix_process_model_to_web_apps/)

best regards and thank you!

**Comment by aikoii** [comment]
👋 @victor-chebotar, it could be because of GitBook's flavored Markdown which may have nuances compared to standard [GFM](https://github.github.com/gfm/) (we try as much as possible to stick to). I'm not entirely sure if this is related to the presence of an underscore in the link or if it's a general issue with how we handle markdown links. If that's okay with you, let's continue the conversation on the email thread you sent to support 🙂

**Comment by victor-chebotar** [comment]
Hi, Aiko!

Thank you for answering!

First, I use vanilla Markdown: link syntax like `[link title](url.com)`

Second, I consider option to work with source files as very essential. Means:

1. GitBook shouldn’t change source files as much as it’s possible.
2. Source files should keep readable and understandable.
3. Don’t invent your own specific syntax.
4. Intention should be clear, when I read specific syntax in source files.

Additional info:

I edited typo in my Bug report https://github.com/orgs/GitbookIO/discussions/368

Link to page: https://app.gitbook.com/o/9AaMpPWyfFJ6QsETZfuC/s/ls0YGZNV1w7SPwolDJtx/arch-architecht-patterns-arkhitektura-patterny-backup/test-markdown-links

How it looks like:
***@***.***
Source file before: Test -- Markdown links.md

Then I committed file and edited it with GitBook UI.
I changed only file description. (description: file edited v1 -> description: file edited v2)

Source file after: Test -- Markdown links v2.md

You can see here that GitBook autoformatted link url.

Best regards, Victor.

**Reply by aikoii** [reply]
Thanks for the insight @victor-chebotar!

So GitBook isn't a direct markdown editor. When editing content in GitBook, we do escape some properties values, the reason why we have our own format is that the editor supports a lot more than just markdown. E.g. all of our blocks can be converted to markdown (with various outputs: https://docs.gitbook.com/content-creation/blocks/hint#representation-in-markdown). While we convert and sync back to markdown, it's not always 1:1. So we use a combination of liquid templates, HTML and markdown with the quirk of, sometimes, encoding things differently.

We'll take note of preserving source files better in the future 🤞. No immediate changes planned, but I appreciate you sharing this feedback!

**Reply by victor-chebotar** [reply]
Thank you for your answer 👍 

And this is the obstacle. I'm researching the option to present GitBook for my company projects, and I can't do this.

1. GitBook **uses its own extended MarkDown syntax**. I can't go back and use another tool without time-consuming migration. It binds the project doc to the tool.
2. Also it **makes silent changes**, that e.g. breaks links.
3. It can **silently delete bits of source code** (see below details)

I discovered that when I uploaded files, GitBook engine **silently completely evicted bits** like this: `<UID>` from plain text.

I don't mean restricting with vanilla Markdown. I like features like e.g. "Hints" 😎 .  But **make it backward-compatible**, please.

**Comment by ykla** [comment]
The issue still persists and continues to trouble me, so I believe you shouldn't mark the current answer as the correct one.

**Comment by ykla** [comment]
GitBook always tries to automatically change my correct markdown tags, and the changes it makes are often incorrect. For example, it always tries to change asterisks * to underscores, which is problematic on GitHub because underscores are not recognized when used with Chinese. How can I prevent GitBook from automatically modifying the format of my text? see <https://github.com/taophilosophy/SEP-CN/commit/03a0f58f15cf4423dd4204026c889328110ec386>
